mne/io/base.py

```diff
@@ -363,6 +368,24 @@ def __init__(self, info, preload=False,
         self._comp = None
         self._filenames = list(filenames)
         self.orig_format = orig_format
+        # Sanity check and set original units, if provided by the reader:
+        if orig_units:
+            ch_l = list(info['ch_names'])
+
+            # STI 014 channel is native only to fif ... for all other formats
+            # this was artificially added by the IO procedure
+            if 'STI 014' in ch_l and not self.filenames[0].endswith('.fif'):
+                ch_l.remove('STI 014')
+
+            # check that channels correspond. note that long ch_names get
+            # truncated, so check first 15 characters
+            orig_units_chs = [ch[:15] for ch in orig_units.keys()]
+            ch_correspond = [ch in orig_units_chs for ch in ch_l]
+            if not all(ch_correspond):
+                ch_without_orig_unit = ch_l[ch_correspond.index(False)]
+                raise ValueError('Channel {0} has no associated original '
+                                 'unit.'.format(ch_without_orig_unit))
```

Comment:
not covered by test

Reply:
where should I place the test? There is no `mne/io/tests/test_base.py`